### PR TITLE
Add naming convention to bundles

### LINKED ISSUE #165: [Admin/Bundles] naming convention of an addon
Add the following description to "Advanced Functionality", wherever fitting.
https://community.openhab.org/t/addons-cfg-proper-way-to-discover-the-name-of-an-addon/17107/3

### PR DESCRIPTION
**Pull request by Confectrician**
- Added a specific Language (due to remark/lint rules) to all code blocks, since I edited the page anyway.
- Added the naming convention paragraph

Closes #165

### CODE REVIEW

File: administration/bundles.md [hunk 1]
```diff
@@ -40,9 +40,26 @@ openhab> bundle:list
...
+```java
+<prefix>-<type>-<id>
+```
+
+where
+
+- **prefix** is ALWAYS "openhab" 
+- **type** is the add-on type, e.g. "binding" or "ui"
+- **id** is the identifier for this bundle
+  (It is also the id that you would have to use openHAB 1.x in addons.cfg, e.g. "hue", "map", etc.)
```
Comment (bgilmer77) [inline administration/bundles.md]: Consider, "This is the id you would have used in the openHAB 1.x addons.cfg file (e.g. "hue", "map", etc.)"

File: administration/bundles.md [hunk 2]
```diff
@@ -40,9 +40,26 @@ openhab> bundle:list
...
+
+  This is used for example to assign them the correct features for the karaf console or to sort them in different tabs within the Paper UI.
```
Comment (bgilmer77) [inline administration/bundles.md]: Consider, "The bundle identifier may be useful to assign the correct features in the karaf console, or to sort the bundle into different tabs within the PaperUI."

If this is not what is meant,  consider revising the text.  The example as written is not clear.

File: administration/bundles.md (hunk 1 above)
Comment (Confectrician) [reply administration/bundles.md]: I think the id is used for many other things internally, so putting the focus only on the 1.x usage could be misunderstood.

@ThomDietrich  wdyt?

File: administration/bundles.md (hunk 2 above)
Comment (Confectrician) [reply administration/bundles.md]: Hm.

The intention was to give at least some examples why the naming convention is existing.
I think automatic sorting in Paper UI is easier to grasp than the Karaf thing.
(Especially for beginners, who are browsing through the docs.)

Maybe i should change it to one Paper UI example and clarify that there are many other usecases.

Wdyt?

Comment (bgilmer77) [reply administration/bundles.md]: Yep - one solid example is enough, and the one about sorting stuff in a UI is easily understood.  Beginners may not yet even be aware... "What is this karaf console of which you speak?".  Boom - gone for an hour of Google searching, only to figure out that, in the early stages, they don't need to know anything about it.

File: administration/bundles.md (hunk 1 above)
Comment (bgilmer77) [reply administration/bundles.md]: I know you ask @ThomDietrich , but my two cents are as follows... "OH1.x??  What's that?".  OH1 is in the rear view mirror, and getting farther away all the time.  At some point we would remove the reference anyway.  I don't see this adds value, and it potentially adds confusion for someone who is just coming on board.

Comment (Confectrician) [reply administration/bundles.md]: I am asking him for a third opinion. :)

After reading your 2 cents my suggestion would be to remove that sentence completely from the beginning.

**Comment by Confectrician** [comment]
Changes done, here we go. :)
Sorry for the delay.

File: administration/bundles.md (hunk 1 above)
Comment (ThomDietrich) [reply administration/bundles.md]: I agree to remove that additional detail all together.

Comment (Confectrician) [reply administration/bundles.md]: Already done in the last commit. :)

**Comment by ThomDietrich** [comment]
Heyho!

Why python? It adds some color but the safe alternative is always `text`.

Regarding the bundle names: The relation doesn't get clear as you didn't mention `feature:list`. Also I believe the first part is not ALWAYS openhab, it's just always openhab for openHAB bundles 😊 Besides that the convention is only half the solution as in many places (e.g. logging) you'll find the package namespace of the bundle. The following command could be mentioned:

```text
openhab> bundle:list -s
START LEVEL 100 , List Threshold: 50
 ID | State    | Lvl | Version                | Name                                                   | Symbolic name
-----------------------------------------------------------------------------------------------------------------------------------------------------------------
206 | Active   |  80 | 2.1.0                  | Exec Binding                                           | org.openhab.binding.exec
207 | Active   |  80 | 1.10.0                 | openHAB Expire Binding                                 | org.openhab.binding.expire
208 | Active   |  80 | 2.1.0                  | Homematic Binding                                      | org.openhab.binding.homematic
209 | Active   |  80 | 1.10.0                 | openHAB MQTT Binding                                   | org.openhab.binding.mqtt
210 | Active   |  80 | 1.10.0                 | openHAB Mqttitude Binding                              | org.openhab.binding.mqttitude
211 | Active   |  80 | 2.1.0                  | Network Binding                                        | org.openhab.binding.network
```

**Comment by Confectrician** [comment]
> Why python?

Because java and evrything else, are doing nearly the same highlighting.
`text` is fone for me too.

> Also I believe the first part is not ALWAYS openhab, it's just always openhab for openHAB bundles 😊

Then my question would be:
Are we going to mention that always is openhab for openHAB bundles or are we going to mention something generic for all kind of bundles?

> The relation doesn't get clear as you didn't mention `feature:list`.
What do you have in mind here?
Just a short sentence that delimits bundles from features?

> Besides that the convention is only half the solution as in many places (e.g. logging) you'll find the package namespace of the bundle. The following command could be mentioned:

Good point, i will add this with your example table.

**Comment by ThomDietrich** [comment]
I'd say let's go with something generic and a second part like "for addons this is often openhab or esh"

Do that, just know that the table is a random grab. maybe use one that matches the other examples?

**Comment by Confectrician** [comment]
Any suggestions for the `feature:list`.
I am not getting something useful in my mind. (Maybe tomorrow. :D)

**Comment by bgilmer77** [comment]
Okay - so question.  I am seeing ```java, ```python, ```text, ```elephant (okay - maybe I made up the elephant part).  Are we going to pick one and normalize everything to that?

**Comment by Confectrician** [comment]
I think we could at least centralize **logging parts** to `text´, that would be an effort.
I use all the other languages on a personal impression of how it looks, when I generate the page.

**Comment by Confectrician** [comment]
> The relation doesn't get clear as you didn't mention `feature:list`.

> What do you have in mind here?
Just a short sentence that delimits bundles from features?

> Any suggestions for the feature:list.
I am not getting something useful in my mind. (Maybe tomorrow. :D)

Still no idea of a useful sentence or paragraph.
I think this should be in one of the first sentences and differentiate between features as an output of all available bindings/bundles and bundles as installed/active features.

But i am really not sure if I got the core here.
Some help and Ideas from your side would be lovely. :)

**Comment by ThomDietrich** [comment]
Hey @Confectrician and @bgilmer77! I'm honestly happy about your recent engagement in the documentation endeavor 🤗 keep up the spirit!

Regarding language indication: First thing we need to remember is, that we want a language for every code block. Our page uses the Kramdown Markdown engine with Rouge as the syntax highlighting component: http://rouge.jneen.net
Now ideally we would use `java` for java code, `python` for python code, and so on...
Sadly many code blocks in documentation are not of one predefined language. openHAB rules for example are constructed based on a custom Xtext DSL. Other snippets like logging lines are simply no code. Parts of code can also be misinterpreted using the wrong language. In these cases we need to look for the best match and that's no science to be honest. In most cases I 'd say "go with the flow". Check the existing configuration articles for examples (Items: java/javascript, Sitemaps: perl, Rules: java or php). As for non-code snippets: Sometimes a programming language might fit but most of the time the non-highlighting option of `text` is the best we can do.

**Comment by ThomDietrich** [comment]
@Confectrician let's merge this now and add further improvements later. If you can summarize the missing aspects, please create an issue! Thanks and Thanks!

**Pull request merged**